### DISCUSSION TITLE
SH-ESP32 Optocoupler input question

### DISCUSSION THREAD

**Discussion by big-col** [hatlabs/discussions#105] [General]
Having read this in the docs:

If you provide a voltage between about 2.5V and 18V in ISO IN, you’ll have the output of the input optocoupler pulled high, meaning that OPTO_IN (GPIO 35) is pulled high.

Does this mean that when i connect the OPTO IN to a 3.3v pin that GPIO35 will change from false to true?

When i use the code below i don't get the expected result, what am i doing wrong please?

 ``` // Create another digital input, this time with RepeatSensor. This approach
  // can be used to connect external sensor library to SensESP!

  const uint8_t kDigitalInput2Pin = 35;
  const unsigned int kDigitalInput2Interval = 3000;

  // Configure the pin. Replace this with your custom library initialization
  // code!
  pinMode(kDigitalInput2Pin, INPUT_PULLUP);```

Ultimately I want to connect the Bilge pump LED to the OPTO IN -,would solution work?

**Comment by mairas** [comment]
Yes, it should... You have the optocoupler GND connected as well, do you?

Try checking the GPIO35 voltage after the optocoupler - either pad of the 0R resistor circled in red here.

<img width="942" alt="Screenshot 2024-03-27 at 10 21 52" src="https://github.com/hatlabs/discussions/assets/407010/099dbf82-414a-4afa-aac5-43f5427d3cee">

**Comment by big-col** [comment]
rechecked connections, now GND is properly connected its working :)

**Comment by mairas** [comment]
Re-opening to provide visibility for future readers.
